File: 0606. Construct String from Binary Tree/Solution0606.py
class Solution0606:
    def tree2str(self, root: Optional[TreeNode]) -> str:
        if not root:
            return ""
        res = str(root.val)
        left = self.tree2str(root.left)
        right = self.tree2str(root.right)
        if left or right:
            res += '('
            res += left
            res += ')'
        if right:
            res += '('
            res += right
            res += ')'
        return res

File: 0606. Construct String from Binary Tree/Solution0606.cpp
#include<string.h>
#include<iostream>
using namespace std;

class Solution0606 {
public:
    string tree2str(TreeNode* root) {
        if (root == nullptr) return "";
        string res = to_string(root->val);
        string left = tree2str(root->left);        
        string right = tree2str(root->right);
        if (left.length() != 0 || right.length() != 0) {
            res += "(";
            res += left;
            res += ")";
        }
        if (right.length() != 0) {
            res += "(";
            res += right;
            res += ")";            
        }
        return res;
    }
};